tests/test_other.py

```diff
@@ -3595,6 +3595,10 @@ def test_dashM_respect_dashO(self):
     assert len(with_dash_o) == 0
     assert len(without_dash_o) != 0
 
+  def test_null_device_as_input(self):
+    out = Popen([PYTHON, EMXX, '-E', '-dM', '-xc', '/dev/null'], stdout=PIPE).communicate()[0]
+    self.assertContained('#define __EMSCRIPTEN__ 1', out) # Verify output contains one of the native predefined macros
```

Comment:
If the null device file was not handled correctly by emcc then it would not produce anything in the stdout, but just error complaining about missing input file (or something like that) in the stderr. So, actually by checking there is some output in stdout and nothing in stderr are sufficient. Here, I went a little bit further to check the output is conforming to "-E -dM" compiler flags. Let me know if you want me to just check "there is output in stdout and nothing in stderr" instead.